### ISSUE TITLE
Can not convert case-insensitive filter to custom expression

### ISSUE THREAD

**Issue by ariya** [metabase/metabase#14959]
### Steps to reproduce

1. Ask a question, simple question
2. Sample Dataset, Reviews
3. Filter by Review, Contains
4. Type `MULLER` in the text input
5. Make sure "Case sensitive" is **not** checked
6. Click _Add Filter_
7. There should only one row in the table. Click on the "Reviewer contains MULLER" filter
8. Click on the left-chevron, scroll down, Custom Expression
9. It should display `=contains([Reviewer], "MULLER")`. Click _Done_

### Expected

There should be still one row in the table

### Actual

No results!

![image](https://user-images.githubusercontent.com/7288/109081108-92907c00-76b6-11eb-8c4b-a4cb9bfd474e.png)

### What happens?

Up to Step 7, the corresponding MBQL is
```
["contains", ["field-id", 9], "MULLER", {case-sensitive: false}]
```

Once that regular filter is converted into a custom expression, the query becomes:

```
["contains", ["field-id", 9], "MULLER"]
```

The conversion from the regular filter to custom expression also *modifies* the underlying filter to be case sensitive, hence there is no match.

**Comment by ariya** [comment]
The same issue with "Starts with" and "Ends with".

Initial MBQL:
```
["starts-with", ["field-id", 9], "GER", {case-sensitive: false}]
```

when converted into custom expression (loss of case-sensitivity flag):
```
["starts-with", ["field-id", 9], "GER"]
```

**Comment by flamber** [comment]
It should only add case handling when the database supports it - example [`driver/supports?`](https://github.com/metabase/metabase/blob/master/modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj#L28)
Related to #12457

**Comment by ariya** [comment]
> It should only add case handling when the database supports it - example [`driver/supports?`](https://github.com/metabase/metabase/blob/master/modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj#L28)

Great catch!

Do we even already have a precedent of (custom expression) front-end code being aware of the database features/limitations?

**Comment by flamber** [comment]
Yes, `regexextract` is one of the things that gets checked - along with others: https://www.metabase.com/docs/latest/users-guide/expressions.html#database-limitations
But it currently doesn't work very well - see issue 12222

**Comment by ariya** [comment]
Looks like generally any filter with options aren't converted to its corresponding custom expression, it says so in `formatFunction`:
```
  if (hasOptions(args)) {
    // FIXME: how should we format args?
    args = args.slice(0, -1);
  }
```

Because of the lack of object format (think of JSON/YAML/end) or key-value map for custom expression, it's challenging to map those options. The best I can come up with is to convert to an optional string, e.g.`Contains([Reviewer], "GERRY", "case-sensitive")` with the current obvious challenges (1) There's no support for optional arguments in function call right now, so that must be implemented (2) what do we do when the value is `false`?
